models/transaction_validation_test.go

```diff
@@ -0,0 +1,181 @@
+package models
+
+import (
+	"encoding/hex"
+	"math/big"
+	"testing"
+
+	"github.com/onflow/go-ethereum/common"
+	"github.com/onflow/go-ethereum/core/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestValidateTransaction(t *testing.T) {
+	t.Parallel()
+
+	t.Run("valid transaction", func(t *testing.T) {
+		t.Parallel()
+
+		to := common.HexToAddress("0x000000000000000000000000000000000000dEaD")
+		tx := types.NewTx(
+			&types.LegacyTx{
+				Nonce:    1,
+				To:       &to,
+				Value:    big.NewInt(0),
+				Gas:      25_000,
+				GasPrice: big.NewInt(0),
+				Data:     []byte{},
+			},
+		)
+
+		err := ValidateTransaction(tx)
+		assert.NoError(t, err)
+	})
+
+	t.Run("send to 0 address", func(t *testing.T) {
+		t.Parallel()
+
+		to := common.HexToAddress("0x0000000000000000000000000000000000000000")
+		tx := types.NewTx(
+			&types.LegacyTx{
+				Nonce:    1,
+				To:       &to,
+				Value:    big.NewInt(0),
+				Gas:      25_000,
+				GasPrice: big.NewInt(0),
+				Data:     []byte{},
+			},
+		)
+
+		err := ValidateTransaction(tx)
+		require.Error(t, err)
+		assert.ErrorContains(t, err, "transaction recipient is the zero address")
+	})
+
+	t.Run("create empty contract (no value)", func(t *testing.T) {
```

Comment:
I suggest making the bellow in a table test: https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
Thanks for the suggestion :raised_hands: 
Reworked in https://github.com/onflow/flow-evm-gateway/pull/314/commits/44b9ccb6b32c29cf87b1fe4e374bf34147f661fc